src/neptune/internal/utils/dependency_tracking.py

```diff
@@ -0,0 +1,47 @@
+__all__ = [
+    "DependencyTrackingStrategy",
+    "InferDependenciesStrategy",
+    "FileDependenciesStrategy",
+]
+
+import os
+import subprocess
+from abc import (
+    ABC,
+    abstractmethod,
+)
+from typing import (
+    TYPE_CHECKING,
+    Union,
+)
+
+from neptune.types import File
+
+if TYPE_CHECKING:
+    from neptune import Run
+
+
+class DependencyTrackingStrategy(ABC):
+    @abstractmethod
+    def log_dependencies(self, run) -> None:
+        ...
+
+
+class InferDependenciesStrategy(DependencyTrackingStrategy):
+    def log_dependencies(self, run: "Run") -> None:
+        try:
+            dependencies_str = subprocess.check_output(["pipreqs", "--print", "."]).decode("utf-8")
+        except subprocess.SubprocessError:
+            return
+
+        if dependencies_str:
+            run["source_code/requirements"].upload(File.from_content(dependencies_str))
```

Comment:
Shouldn't we add `extension='.txt'`?

Reply:
I think it's the default if the argument of `from_content` is a string